camel/models/interlm_xcomposer_model.py

```diff
@@ -0,0 +1,325 @@
+        super().__init__(
+            model_type,
+            model_config_dict,
+            api_key=None,
+            url=None,
+            token_counter=None,
+        )
+
+        from pathlib import Path
+
+        self.multimodal_src_directory: Path = Path(
+            model_config_dict.get("multimodal_src_path", "./examples/src")
+        )
+
+        # `model_path` is the path to the open-source model
+        model_path: Optional[str] = self.model_config_dict.get(
+            "model_path", None
+        )
+        if not model_path:
+            raise ValueError("Path to open-source model is not provided.")
+        self.model_path: str = model_path
+
+        self.model_name: str = self.model_path.split('/')[-1]
+
+        try:
+            # `operation_mode` is the operation mode of the open-source model
+            # Refer to https://github.com/InternLM/InternLM-XComposer
+            self.operation_mode = model_config_dict.get(
+                "operation_mode", OperationMode.CHAT
+            )
+        except ValueError:
+            valid_modes = [mode.value for mode in OperationMode]
+            raise ValueError(
+                "Invalid operation mode for open-source model "
+                f"backend. Choose from {valid_modes}"
+            )
+
+        import torch
+
+        model_kwargs = self.model_config_dict.get("model_kwargs", {})
+        model_kwargs["torch_dtype"] = torch.bfloat16
+        model_kwargs["trust_remote_code"] = True
+
+        from transformers import AutoModel, AutoTokenizer
+
+        self._client = (
+            AutoModel.from_pretrained(
+                self.model_name,
+                **model_kwargs,
+            )
+            .to(
+                model_kwargs.get("device") or "cuda"
+                if torch.cuda.is_available()
+                else "cpu"
+            )  # move the model to the specified device (GPU or CPU)
+            .eval()  # set the model to evaluation mode
+            .half()  # convert the model parameters to half-precision floating point (FP16).  # noqa: E501
+        )
+
+        tokenizer_kwargs = self.model_config_dict.get("tokenizer_kwargs", {})
+        tokenizer_kwargs["trust_remote_code"] = True
+
+        self._tokenizer = AutoTokenizer.from_pretrained(
+            self.model_name, **tokenizer_kwargs
+        )
+        self._client.tokenizer = self._tokenizer  # mypy: ignore[attr-defined]
+
+        # Replace `model_config_dict` with only the `model_kwargs`
+        self.model_config_dict = self.model_config_dict["model_kwargs"]
+
+    @property
+    def token_counter(self) -> BaseTokenCounter:
+        r"""Initialize the token counter for the model backend.
+
+        Returns:
+            BaseTokenCounter: The token counter following the model's
+                tokenization style.
+        """
+        if not self._token_counter:
+            self._token_counter = OpenAITokenCounter(ModelType.GPT_3_5_TURBO)
+        return self._token_counter
```

Comment:
Why we use `OpenAITokenCounter(ModelType.GPT_3_5_TURBO)` for multi-modal models?